File: pmcjs/scripts/xmlparser.pl

use XML::Parser;
my $parser = XML::Parser->new();

sub handle_text() {
  print "text: $_[1]\n";
}

my $handle_default = sub {
  print "default: $_[1]\n";
};

$parser->setHandlers('Char', \&handle_text);
$parser->setHandlers('Default', $handle_default);
$parser->parsefile("domtest.xml");

File: pmcjs/samples/xmlparser.js

use("Perl");
Perl.Use("XML::Parser");
var parser = new Perl("XML::Parser");

function handle_start(expat, elem, attr, val) {
  print("start: [" + elem + "]");
}
function handle_end(expat, elem) {
  print("end: [" + elem + "]");
}
function handle_text(expat, str) {
  print("char: [" + str + "]");
}
function handle_default(expat, str) {
  print("default: [" + str + "]");
}

parser.setHandlers('Char', handle_text);
parser.setHandlers('Start', handle_start);
parser.setHandlers('End', handle_end);
parser.setHandlers('Default', handle_default);
parser.parsefile("domtest.xml");

